From 1dc34d3d0c88f1fcf3fb9144437633295a40e7b8 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 30 May 2024 11:35:57 +0200
Subject: [PATCH 0849/1695] dt-bindings: regulator: add STM32MP21 regulator
 bindings

Add STM32MP21 specific regulator bindings, for SMCI
- SOC Internal regulators
- STPMIC regulators, with all supported variant:
  STPMIC25/STM32MP2L/STPMIC1L
- External regulators, for example for GPIO regulator

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 880f363216e5f9d6986adca9f9c2493c1d642742 dt-bindings: regulator: add STM32MP21 regulator bindings]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I0947216924f820abd682b54dea9ce9b13ad647b1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/383939
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../regulator/st,stm32mp21-regulator.h        | 48 +++++++++++++++++++
 1 file changed, 48 insertions(+)
 create mode 100644 core/include/dt-bindings/regulator/st,stm32mp21-regulator.h

diff --git a/core/include/dt-bindings/regulator/st,stm32mp21-regulator.h b/core/include/dt-bindings/regulator/st,stm32mp21-regulator.h
new file mode 100644
index 000000000..aac642636
--- /dev/null
+++ b/core/include/dt-bindings/regulator/st,stm32mp21-regulator.h
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * Copyright (C) 2024, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef __DT_BINDINGS_REGULATOR_STM32MP21_REGULATOR_H
+#define __DT_BINDINGS_REGULATOR_STM32MP21_REGULATOR_H
+
+/* SCMI voltage domains identifiers */
+
+/* SOC Internal regulators */
+#define VOLTD_SCMI_VDDIO1		0
+#define VOLTD_SCMI_VDDIO2		1
+#define VOLTD_SCMI_VDDIO3		2
+#define VOLTD_SCMI_ADC			3
+#define VOLTD_SCMI_VREFBUF		4
+
+/* STPMIC regulators (STPMIC25/STM32MP2L/STPMIC1L) */
+#define VOLTD_SCMI_STPMIC_BUCK1		5
+#define VOLTD_SCMI_STPMIC_BUCK2		6
+#define VOLTD_SCMI_STPMIC_BUCK3		7
+#define VOLTD_SCMI_STPMIC_BUCK4		8
+#define VOLTD_SCMI_STPMIC_BUCK5		9
+#define VOLTD_SCMI_STPMIC_BUCK6		10
+#define VOLTD_SCMI_STPMIC_BUCK7		11
+#define VOLTD_SCMI_STPMIC_LDO1		12
+#define VOLTD_SCMI_STPMIC_LDO2		13
+#define VOLTD_SCMI_STPMIC_LDO3		14
+#define VOLTD_SCMI_STPMIC_LDO4		15
+#define VOLTD_SCMI_STPMIC_LDO5		16
+#define VOLTD_SCMI_STPMIC_LDO6		17
+#define VOLTD_SCMI_STPMIC_LDO7		18
+#define VOLTD_SCMI_STPMIC_LDO8		19
+#define VOLTD_SCMI_STPMIC_REFDDR	20
+#define VOLTD_SCMI_STPMIC_GPO1		21
+#define VOLTD_SCMI_STPMIC_GPO2		22
+#define VOLTD_SCMI_STPMIC_GPO3		23
+#define VOLTD_SCMI_STPMIC_GPO4		24
+#define VOLTD_SCMI_STPMIC_GPO5		25
+
+/* External regulators */
+#define VOLTD_SCMI_REGU0		26
+#define VOLTD_SCMI_REGU1		27
+#define VOLTD_SCMI_REGU2		28
+#define VOLTD_SCMI_REGU3		29
+#define VOLTD_SCMI_REGU4		30
+
+#endif /*__DT_BINDINGS_REGULATOR_STM32MP21_REGULATOR_H */
-- 
2.47.2

